src/components/diagrams/diagram.js

```diff
@@ -0,0 +1,870 @@
+                        setModificationInProgress(value)
+                    }
+                />
+            )
+        );
+    };
+
+    const displayMenu = (equipmentType, menuId) => {
+        const Menu = withEquipmentMenu(
+            BaseEquipmentMenu,
+            menuId,
+            equipmentType
+        );
+        return (
+            equipmentMenu.display &&
+            equipmentMenu.equipmentType === equipmentType && (
+                <Menu
+                    id={equipmentMenu.equipmentId}
+                    position={equipmentMenu.position}
+                    handleClose={closeEquipmentMenu}
+                    handleViewInSpreadsheet={handleViewInSpreadsheet}
+                />
+            )
+        );
+    };
+
+    let sizeWidth,
+        sizeHeight = initialHeight;
+    if (svg.error) {
+        sizeWidth = errorWidth;
+    } else if (finalPaperWidth != null && finalPaperHeight != null) {
+        sizeWidth = finalPaperWidth;
+        sizeHeight = finalPaperHeight;
+    } else if (initialWidth !== undefined || loadingState) {
+        sizeWidth = initialWidth;
+    } else {
+        sizeWidth = props.totalWidth; // happens during initialization if initial width value is undefined
+    }
+
+    if (!isNodeBuilt(currentNode)) {
+        sizeWidth = props.totalWidth / props.numberToDisplay; // prevents the diagram from becoming too big if the current node is not built
+    }
+
+    if (sizeWidth !== undefined) {
+        initialWidth = sizeWidth; // setting initial width for the next SLD.
+    }
+    if (sizeHeight !== undefined) {
+        initialHeight = sizeHeight; // setting initial height for the next SLD.
+    }
+
+    if (!fullScreenDiagram?.id && props.computedHeight) {
+        sizeHeight = props.computedHeight;
+    }
+
+    /**
+     * DIAGRAM CONTROL HANDLERS
+     */
+
+    const onMinimizeHandler = () => {
+        minimizeDiagramView(props.diagramId, props.svgType);
+        dispatch(setFullScreenDiagram(null));
+    };
+
+    const onTogglePinHandler = () => {
+        togglePinDiagramView(props.diagramId, props.svgType);
+    };
+
+    const onCloseHandler = () => {
+        dispatch(setFullScreenDiagram(null));
+        closeDiagramView(props.diagramId, props.svgType);
+        if (props.svgType === SvgType.NETWORK_AREA_DIAGRAM) {
+            dispatch(resetNetworkAreaDiagramDepth());
+        }
+    };
+
+    const onShowFullScreenHandler = () => {
+        dispatch(setFullScreenDiagram(props.diagramId, props.svgType));
+    };
+
+    const onHideFullScreenHandler = () => {
+        dispatch(setFullScreenDiagram(null));
+    };
+
+    const onIncrementDepthHandler = () => {
+        dispatch(incrementNetworkAreaDiagramDepth());
+    };
+
+    const onDecrementDepthHandler = () => {
+        dispatch(decrementNetworkAreaDiagramDepth());
+    };
+
+    /**
+     * RENDER
+     */
+
+    return !svg.error ? (
+        <Paper
+            ref={ref}
```

Comment:
I don't think this is right

Reply:
After testing you are correct, I removed it.